mlflow/java/src/test/java/org/mlflow/MLeapPredictorTest.java

```diff
@@ -0,0 +1,86 @@
+package org.mlflow.sagemaker;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+import ml.combust.mleap.runtime.frame.Transformer;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mlflow.MLflowRootResourceProvider;
+import org.mlflow.mleap.MLeapLoader;
+import org.mlflow.utils.SerializationUtils;
+
+/** Unit tests for the {@link MLeapPredictor} */
+public class MLeapPredictorTest {
+  @Test
+  public void testMLeapPredictorGetPipelineYieldsValidMLeapTransformer()
+      throws PredictorLoadingException {
+    String modelPath = MLflowRootResourceProvider.getResourcePath("mleap_model");
+    MLeapPredictor predictor = (MLeapPredictor) (new MLeapLoader()).load(modelPath);
```

Comment:
Er, how does this work? MLeapLoader (in MLeap's package?) constructs an instance of our `MLeapPredictor`?

Reply:
Yes. The alternative is to invoke the `MLeapPredictor` constructor with the model artifact path and schema path.

Reply:
How does `MLeapLoader` know how to construct our version of the predictor? What kind of reflection black magic is going on?

Reply:
The `MLeapLoader.load()` method is the inherited method `LoaderModule.load()`. This method loads the configuration for the model specified by `modelPath`, parses the flavor configuration defined with the overall model configuration, and calls `MLeapLoader.createPredictor()` method with the model's root path (`modelRootPath=modelPath`) and the flavor configuration as arguments. `MLeapLoader.createPredictor()` is an implementation of the abstract `LoaderModule.createPredictor()` method that returns an `MLeapPredictor` object. No reflection occurs.

Reply:
Ah, sorry, I didn't realize `MLeapLoader` was a class in our repo (mis-saw it as `org.mleap`). We should change the return type to `MLeapPredictor` directly; I believe in Java you can return more specific return types.

Reply:
Changed the return type of `MLeapLoader.createPredictor()` to `MLeapPredictor`. Unfortunately, I don't see a good way to specialize the return type of `MLeapPredictor.load()` beyond `Predictor` without overriding the super class `LoaderModule.load()` methods.